orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -382,7 +433,49 @@ static Status ModifyParametersForOptimizerPartitioning(
       new_gradient_argdefs.push_back(gradient_argdef);
     }
 
-    offset += tensor_count;
+    offset += tensor_bytes;
+  }
+
+  // weight padding
+  const size_t padding_bytes = padded_buffer_bytes - total_bytes;
+  const int64_t padding_size = padding_bytes / element_size;
+
+  std::cout << "padded_buffer_bytes " << padded_buffer_bytes << " total_bytes " << total_bytes << "\n";
+  std::cout << "padding_bytes " << padding_bytes << " padding_size " << padding_size << "\n";
+
+  if (padding_size > 0) {
```

Comment:
only the last weight will have padding, right? can we add a check here.

Reply:
oh seems you did not align the whole buffer with 256. We are still align 256 for each weight?